extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -15,6 +15,67 @@ const BackbeatConsumer = require('../../../lib/BackbeatConsumer');
 const BackbeatClient = require('../../../lib/clients/BackbeatClient');
 const QueueEntry = require('../utils/QueueEntry');
 
+class _UserAuthManager {
+    constructor(authConfig, log) {
+        assert.strictEqual(authConfig.type, 'user');
+
+        this.log = log;
+        const userInfo = authdata.users.find(
+            user => user.name === authConfig.user);
+        if (userInfo === undefined) {
+            throw Error(`No such user registered: ${authConfig.user}`);
+        }
+        if (userInfo.arn === undefined) {
+            throw Error(`Configured user ${authConfig.user} has no ` +
+                        '"arn" property defined');
+        }
+        if (userInfo.canonicalID === undefined) {
+            throw Error(`Configured user ${authConfig.user} has no ` +
+                        '"canonicalID" property defined');
+        }
+        if (userInfo.displayName === undefined) {
+            throw Error(`Configured user ${authConfig.user} has no ` +
+                        '"displayName" property defined');
+        }
+        this.userArn = userInfo.arn;
+        this.canonicalID = userInfo.canonicalID;
+        this.displayName = userInfo.displayName;
+        this.credentialProvider = new AWS.CredentialProviderChain([
+            new AWS.Credentials(userInfo.keys.access,
+                                userInfo.keys.secret),
+        ]);
+    }
+
+    getCredentialProvider() {
+        return this.credentialProvider;
+    }
+
+    lookupUserAttributes(userId, cb) {
+        if (!this.userArn.startsWith(userId)) {
+            this.log.error('Target user for replication must match ' +
+                           'configured destination user ARN',
+                           { targetUser: userId,
+                             localUserArn: this.userArn });
+            return process.nextTick(() => cb(errors.NotImplemented));
+        }
+        // return local user's attributes
+        return process.nextTick(
+            () => cb(null, { canonicalID: this.canonicalID,
+                             displayName: this.displayName }));
+    }
+}
+
+class _RoleAuthManager {
+    constructor() {
+        throw Error('Role-based authentication not yet implemented');
+    }
+}
+
+
+function _extractUserFromRole(role) {
+    return role.split(':').slice(0, 5).join(':');
```

Comment:
This gives back the same string, I am not sure if this is intentional. Also, role arn contains accountId not user.